jicofo/src/main/kotlin/org/jitsi/jicofo/conference/Participant.kt

```diff
@@ -0,0 +1,375 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015-Present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.conference
+
+import org.jitsi.impl.protocol.xmpp.ChatRoomMember
+import org.jitsi.jicofo.ConferenceConfig
+import org.jitsi.jicofo.TaskPools.Companion.scheduledPool
+import org.jitsi.jicofo.conference.source.ConferenceSourceMap
+import org.jitsi.jicofo.conference.source.VideoType
+import org.jitsi.jicofo.discovery.DiscoveryUtil
+import org.jitsi.jicofo.util.Cancelable
+import org.jitsi.jicofo.xmpp.jingle.JingleSession
+import org.jitsi.jicofo.xmpp.muc.SourceInfo
+import org.jitsi.jicofo.xmpp.muc.hasModeratorRights
+import org.jitsi.utils.OrderedJsonObject
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.LoggerImpl
+import org.jitsi.xmpp.extensions.jingle.JingleIQ
+import org.jxmpp.jid.EntityFullJid
+import org.jxmpp.jid.Jid
+import java.time.Clock
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.Deque
+import java.util.LinkedList
+import java.util.Locale
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+/**
+ * Class represent Jitsi Meet conference participant. Stores information about
+ * Colibri channels allocated, Jingle session and media sources.
+ *
+ * @author Pawel Domas
+ * @author Boris Grozev
+ */
+open class Participant @JvmOverloads constructor(
+    /** The associated chat member. */
+    val chatMember: ChatRoomMember,
+    /** The conference in which this participant participates. */
+    private val conference: JitsiMeetConferenceImpl,
+    parentLogger: Logger? = null,
+    /** The list of XMPP features supported by this participant. */
+    private val supportedFeatures: List<String> = DiscoveryUtil.getDefaultParticipantFeatureSet(),
+    /** The [Clock] used by this participant. */
+    private val clock: Clock = Clock.systemUTC()
+) {
+    /** The endpoint ID for this participant in the videobridge (Colibri) context. */
+    val endpointId: String = chatMember.name
```

Comment:
Is `name` definitely non-volatile in `ChatRoomMember`?  I wonder if implementing this as a `get()` might be safer.

Reply:
Yes, it's a `final` set to the JID's resource part. "Name" is a horrible name, but that's for another PR.